csync/src/std/c_utf8.cc

```diff
@@ -64,26 +60,33 @@ char* c_utf8_from_locale(const mbchar_t *wstr)
     WideCharToMultiByte(CP_UTF8, 0, wstr, len, mdst, size_needed, NULL, NULL);
     dst = mdst;
   }
+  return dst;
 #else
-  dst = c_strdup(wstr);
+    QTextDecoder dec(QTextCodec::codecForLocale());
+    QString s = dec.toUnicode(wstr, qstrlen(wstr));
```

Comment:
's' can't be const for two reason:  First, on mac, we update it with `s = s.normalized(QString::NormalizationForm_C);`  
Second, we std::move it before calling toUtf8. This saves an allocation as there is an overload of toUtf8 which does the conversion in-place. (but does not work if s was const)